File: solidprinciplesjavakotlin/src/main/java/com/mfcwl/app/solidprinciples/solidprincipleinjava/openclosedprinciple/solution/shapes/Rectangle.java
package com.mfcwl.app.solidprinciples.solidprincipleinjava.openclosedprinciple.solution.shapes;

import com.mfcwl.app.solidprinciples.solidprincipleinjava.openclosedprinciple.solution.interfacedetails.GeometricFigure;

public class Rectangle implements GeometricFigure {

    private final double width;
    private final double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public double calculateArea() {
        return this.width * this.height;
    }
}

File: solidprinciplesjavakotlin/src/main/java/com/mfcwl/app/solidprinciples/solidprincipleinkotlin/openclosedprinciple/notfollowingprinciple/Rectangle.kt
package com.mfcwl.app.solidprinciples.solidprincipleinkotlin.openclosedprinciple.notfollowingprinciple

class Rectangle {
    private var length: Double = 0.0
    private var height: Double = 0.0

    fun getArea(): Double{
        return length*height
    }
}